src/js/base/model.js

```diff
@@ -105,7 +105,22 @@ export default Backbone.Model.extend(extend({
 
     return isFunction(messages[category]) ? messages[category] : this[messages[category]];
   },
-  handleMessage({ category, resource, author, payload }) {
+  queueMessage(data) {
+    if (!this.notifications) {
+      this.notifications = [data];
+      return;
+    }
```

Comment:
Is there a way to set this automatically set model.notifications: [] initially for each model? Do we want to do that?

If we did that we wouldn't have to check if this.notifications exists here. And just do the push().

Reply:
And wasn't sure what to use for naming. But went with model.notifications for now.

Reply:
there is not a good place to instantiate this, but it should probably be `_notifications` and it should be `{}`. where `this.notifications['StateChanged']` would always be the latest `StateChanged`

Reply:
`_notifications` naming makes sense.

But why an object `{}` instead of an array `[{}]`?

Reply:
because we only need to run the most recent of a particular `category`

Reply:
Ah okay, right. Using an object we can just replace the previous notification in the queue if one already exists with the same notification category (`StateChanged`, `OwnerChanged`, etc.).